Add Flatpak build files and docs

janvhs opened the pull request:
This patch adds documentation on updating the bindings oneself, and therefore closes #5 and possibly #3.

Furthermore, it adds a way to generate the bindings using Flatpak.
This comes in handy, if one wants to target Flatpak as a build target. Using this build target, one has the newest version of Libadwaita available to them.

janvhs committed:
Add Flatpak build files and docs

This patch adds documentation on updating the bindings oneself and therefore closes #5 and possibly #3.

Furthermore, it adds a way to generate the bindings using Flatpak.
This comes in handy, if one wants to target Flatpak as a build target.
Using this build target, one has the newest version of Libadwaita available to them.

janvhs commented:
Oh, I forgot to include a copy step to the source tree, after building with Flatpak.
I am going to add that later this day or tomorrow, if you're interested in supporting Flatpak.

diamondburned commented:
> I am going to add that later this day or tomorrow, if you're interested in supporting Flatpak.

Do you still want this PR merged right now?

diamondburned commented on README.md:
```diff
@@ -2,6 +2,57 @@
 
 [adw]: https://gnome.pages.gitlab.gnome.org/libadwaita/doc/
+[flatpak]: https://docs.flatpak.org/en/latest/building-introduction.html
+[gotk4-contributing]: https://github.com/diamondburned/gotk4/blob/70c1e83a63e4f4c18ad89baedd2e726d2052c26f/CONTRIBUTING.md
```
Please bump this commit or link directly to the branch. This file should forever be present in the repository.

janvhs commented:
No, I would have put it on draft, if GitHub allowed for that.
Gonna send you a comment when I am done :)

janvhs committed:
Update the generated Go code when building with Flatpak

Currently Flatpak would not move the generated Go code from .flatpak-builder/build to the original source.
Therefore, `--run` is used to run the generation on the current source tree.

As requested, the link for Contributing is updated.

janvhs commented:
Oke, now it is ready to be merged :smile_cat:

Do you mind if I shoot you a message regarding gotk4 on Mastodon or Matrix, if you're using that?

diamondburned commented:
> Do you mind if I shoot you a message regarding gotk4 on Mastodon or Matrix, if you're using that?

Matrix would be slightly easier than Mastodon for me.
